contrib/scripts/unloaddb.sh

```diff
@@ -0,0 +1,402 @@
+}
+
+#
+# Kill unloaddb processes in progress and delete the directory, and delete incomplete files
+# Directories/files created by the normally terminated unloaddb process is not deleted.
+#
+function cleanup ()
+{
+        local i
+        local pid
+        local exit_stat
+
+	echo "interrupted"
+
+        for ((i = 0; i < $num_proc; i++))
+        do
+                if [ -f $database.$i/unloaddb_$i.pid ];then
+                        pid=$(cat $database.$i/unloaddb_$i.pid)
+
+                        kill -0 $pid 2> /dev/null
+                        if [ $? -eq 0 ];then
+                                kill -9 $pid
+                                rm -rf $database.$i
+                        fi
+                fi
+        done
+}
+
+function get_password ()
+{
+	local password
+
+	read -sp "Enter Password : " password
+	echo $password
+	echo "" > /dev/tty
+}
+
+function verify_user_pass ()
+{
+	local msg
+	local USERNAME
+	local dbuser
+	local passwd
+	local qry_dba_grp="SELECT u.name FROM db_user AS u, TABLE(u.groups) AS g(x) where x.name = 'DBA'"
+	local retcode
+
+	dbuser=$(echo ${user} | cut -d' ' -f2-2)
+	USERNAME=$(echo ${dbuser} | tr [:lower:] [:upper:])
+
+	# check the database server is running
+	retcode=$(ps -ef | grep cub_server | grep $database | wc -l)
+
+	if [ $retcode -eq 0 ];then
+		echo "Database server '$database' is not running"
+		exit 1
+	fi
+
+	dba_groups=$(csql -u public -l -c "$qry_dba_grp" $database | grep -w $USERNAME | wc -l)
+
+	if [ $USERNAME != "DBA" ] && [ $dba_groups -eq 0 ];then
+		echo "User '$dbuser' is not a member of DBA group"
+		exit 2
+	fi
+
+	# Try with NULL password
+	passwd=$(csql $user --password="" -c "SELECT 1" $database 2> /dev/null)
+
+	if [ $? -ne 0 ];then
+		passwd=$(get_password)
+		pass="-p $passwd"
+
+		passwd=$(csql $user $pass -c "SELECT 1" $database 2> /dev/null)
+		if [ $? -ne 0 ];then
+			echo "$dbuser: Incorrect or missing password"
+			exit 4
+		fi
+	fi
+}
+
+function get_options ()
+{
+         while getopts ":D:u:i:t:v" opt; do
+                case $opt in
+                        u ) user="-u $OPTARG" ;;
+                        i ) from_file="$OPTARG" ;;
+                        t ) num_proc="$OPTARG" ;;
+                        D ) target_dir="$OPTARG" ;;
+                        v ) verbose="yes" ;;
+                esac
+        done
+}
+
+function silent_cd ()
+{
+        cd $* > /dev/null
+}
+
+function extract_db_name ()
+{
+        local nelem=$#
+        local i
+
+        # extract last argument & set as database name
+
+        if [ $nelem -eq 0 ];then
+                show_usage
+                exit
+        fi
+
+        database=$(echo ${@:$nelem:1})
+}
+
+function get_table_size ()
+{
+        local num_rows=0
+        local table_name=$1
+        local row_size=0
+        local table_size=0
+
+        num_rows=$(csql $user $pass -l -c "show heap capacity of $table_name" $db | grep Num_recs | awk '{print $3}')
+        row_size=$(get_schema_size $table_name)
```

Comment:
Since we can get  avg_rec_len of the table from "show heap capacity of", 
how about replacing get_schema_size function to avg_rec_len value ?

Reply:
I saw that parameter of csql output but not sure it is correct answer for record length.
I accept your recommendation and change table size as (two args are output of "**csql** show heap capacity of _table_")
size of **table** = **Num_recs** * **Avg_rec_len**